web/src/components/query-summary/summary-panel.tsx

```diff
@@ -291,6 +293,41 @@ export const SummaryPanelContent: React.FC<{
     ) : undefined;
   };
 
+  const dnsLatency = (filteredFlows: Record[]) => {
+    const filteredDNSFlows = filteredFlows.filter(f => f.fields.DnsLatencyMs !== undefined);
+
+    const dnsLatency = filteredDNSFlows.length
+      ? filteredDNSFlows.map(f => f.fields.DnsLatencyMs!).reduce((a, b) => a + b, 0) / filteredDNSFlows.length
+      : NaN;
+
+    return (
+      <Text className="summary-config-item">{`${t('DNS latency')}: ${
+        isNaN(dnsLatency) ? t('n/a') : formatDurationAboveMillisecond(dnsLatency)
+      }`}</Text>
+    );
+  };
+
+  const timeContent = () => {
+    const filteredFlows = flows || [];
+    const duration =
```

Comment:
I guess a check is needed that `filteredFlows.length` is >0 (avoid div by 0)

Reply:
here we go: https://github.com/netobserv/network-observability-console-plugin/pull/324/commits/3cf1597b83ee456a7abeeebcbf04a6296a426993